pybamm/spatial_methods/finite_volume.py

```diff
@@ -142,20 +142,15 @@ def divergence(self, symbol, discretised_symbol, boundary_conditions):
             discretised_symbol = pybamm.NumpyConcatenation(lbc, discretised_symbol, rbc)
 
         domain = symbol.domain
-        # check for spherical domains
+        # check for particle domain
+        divergence_matrix = self.divergence_matrix(domain)
         if ("negative particle" or "positive particle") in domain:
```

Comment:
This is quite specific to the domain names. Can the check be on the SpatialVariable (i.e. is it "x/y/z" or "r")?